tests/UnifiedSpecTests/UnifiedSpecTest.php

```diff
@@ -0,0 +1,283 @@
+                $testArgs[$name] = [$test, $schemaVersion, $runOnRequirements, $createEntities, $initialData];
+            }
+        }
+
+        return $testArgs;
+    }
+
+    /**
+     * Checks server version and topology requirements.
+     *
+     * @param array $runOnRequirements
+     * @throws SkippedTest unless one or more runOnRequirements are met
+     */
+    private function checkRunOnRequirements(array $runOnRequirements)
+    {
+        $this->assertNotEmpty($runOnRequirements);
+
+        $serverVersion = $this->getCachedServerVersion();
+        $topology = $this->getCachedTopology();
+
+        foreach ($runOnRequirements as $o) {
+            $runOnRequirement = new RunOnRequirement($o);
+            if ($runOnRequirement->isSatisfied($serverVersion, $topology)) {
+                return;
+            }
+        }
+
+        $this->markTestSkipped(sprintf('Server version "%s" and topology "%s" do not meet test requirements', $serverVersion, $topology));
+    }
+
+    /**
+     * Return the server version (cached for subsequent calls).
+     *
+     * @return string
+     */
+    private function getCachedServerVersion()
+    {
+        static $cachedServerVersion;
+
+        if (isset($cachedServerVersion)) {
+            return $cachedServerVersion;
+        }
+
+        $cachedServerVersion = $this->getServerVersion();
+
+        return $cachedServerVersion;
+    }
+
+    /**
+     * Return the topology type (cached for subsequent calls).
+     *
+     * @return string
+     * @throws UnexpectedValueException if topology is neither single nor RS nor sharded
+     */
+    private function getCachedTopology()
+    {
+        static $cachedTopology = null;
+
+        if (isset($cachedTopology)) {
+            return $cachedTopology;
+        }
+
+        switch ($this->getPrimaryServer()->getType()) {
+            case Server::TYPE_STANDALONE:
+                $cachedTopology = RunOnRequirement::TOPOLOGY_SINGLE;
+                break;
+
+            case Server::TYPE_RS_PRIMARY:
+                $cachedTopology = RunOnRequirement::TOPOLOGY_REPLICASET;
+                break;
+
+            case Server::TYPE_MONGOS:
+                $cachedTopology = $this->isShardedClusterUsingReplicasets()
+                    ? RunOnRequirement::TOPOLOGY_SHARDED_REPLICASET
+                    : RunOnRequirement::TOPOLOGY_SHARDED;
+                break;
+
+            default:
+                throw new UnexpectedValueException('Toplogy is neither single nor RS nor sharded');
+        }
+
+        return $cachedTopology;
+    }
+
+    /**
+     * Checks is a test format schema version is supported.
+     *
+     * @param string $schemaVersion
+     * @return boolean
+     */
+    private function isSchemaVersionSupported($schemaVersion)
+    {
+        if (version_compare($schemaVersion, '1.0', '>=') && version_compare($schemaVersion, '1.1', '<')) {
+            return true;
+        }
+
+        return false;
```

Comment:
Suggesting a shorter alternative similar to `isShardedCluster`:

```suggestion
        return version_compare($schemaVersion, '1.0', '>=') && version_compare($schemaVersion, '1.1', '<');
```

Reply:
SGTM. This was written with the expectation that we might one day support multiple ranges (e.g. 1.0-1.5, 2.0-2.3), but that's a long way off.